test/Dialect/FIRRTL/prefix-modules.mlir

```diff
@@ -287,3 +287,48 @@ firrtl.circuit "GCTDataMemTapsPrefix" {
     firrtl.instance dut @DUT2()
   }
 }
+
+// Test the the NonLocalAnchor is properly updated.
+// CHECK-LABEL: firrtl.circuit "FixNLA" {
```

Comment:
Test looks great!

One extreme nit... when there are multiple CHECK statements back-to-back, it's easier to read them if you align/indent. E.g.:

```
CHECK:      firrtl.module @Foo
CHECK-NEXT:   %w = firrtl.wire
```